wire/internal/wire/analyze.go

```diff
@@ -204,7 +204,7 @@ dfs:
 			})
 		case pv.IsValue():
 			v := pv.Value()
-			if !types.Identical(v.Out, curr.t) {
+			if !types.Identical(pv.ConcreteType(), curr.t) {
```

Comment:
Keep using `v.Out` for this one since it is unambiguous and to keep consistency with the next few lines.

Reply:
Done.